app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,58 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: [:index]
+
+  def index
+    @received_requests = current_user.friend_requests_received.pending
```

Comment:
I don't think this is the proper place we should place this logic. Our `post_controller` 's `index` action has to have something to do with the `posts`. We probably are giving too much responsibility to it.